core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.resource
+
+/**
+ * An executor resource request. This is used in conjuntion with the ResourceProfile to
+ * programmatically specify the resources needed for an RDD that will be applied at the
+ * stage level.
+ * There are alternative constructors for working with Java.
+ *
+ * @param resourceName Name of the resource
+ * @param amount Amount requesting
+ * @param units Units of amount for things like Memory, default is no units, only byte
+ *              types (b, mb, gb, etc) are supported.
+ * @param discoveryScript Script used to discovery the resources
```

Comment:
I'd love to see more about this. I'm a bit confused about how we expect folks to use this. I was assuming we'd maybe ask the cluster manager for resources but this seems to indicate a very different approach. Is this because cluster managers don't have a consistent way of keeping track of the acceleration resources?

Reply:
not all cluster managers  tell you what addresses for the accelerators they have allocated to you so the discovery Script adds the ability to find them.  If you aren't running in an isolated environment you can also envision someone writing a discovery script that handles getting an accelerator not being used by another process.

Reply:
the discovery Script is really part of the original accelerator aware scheduling, it just carries over to this stage level api.  You can find the docs on that in: https://issues.apache.org/jira/browse/SPARK-27495

Reply:
So there isn't a lot about the discovery script in 27495 or 24615 but I get that we've made that decision. I still think we should clarify it a bit more (and I think you've done this).